### PR TITLE
[TIMOB-24798] Fixed Xcode 9 support.

### PR DESCRIPTION
**Pull request by cb1kenobi**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-24798

### CODE REVIEW

File: iphone/cli/commands/_build.js [hunk 1]
```diff
@@ -2374,6 +2348,30 @@ iOSBuilder.prototype.initialize = function initialize() {
 	}
 };
 
+iOSBuilder.prototype.findProvisioningProfile = function findProvisioningProfile(target, uuid) {
+	var provisioning = this.iosInfo.provisioning;
+
+	function getPP(type, uuid) {
+		var list = provisioning[type];
+		for (var i = 0, l = list.length; i < l; i++) {
```
Comment (janvennemann) [inline iphone/cli/commands/_build.js]: A for..of loop would be easier to read

File: iphone/cli/hooks/package.js [hunk 2]
```diff
@@ -174,40 +174,44 @@ exports.init = function (logger, config, cli) {
...
+			if (pp.appPrefix) {
+				exportsOptions.teamId = pp.appPrefix;
 			}
 		}
 
+		var keychains = builder.iosInfo.certs.keychains;
+		Object.keys(keychains).some(function (keychain) {
+			return (keychains[keychain].distribution || []).some(function (d) {
+				if (!d.invalid && d.name === builder.certDistributionName) {
+					exportsOptions.signingCertificate = d.fullname;
```
Comment (janvennemann) [inline iphone/cli/hooks/package.js]: Missing return so `some` won't break early

**Review by janvennemann — CHANGES_REQUESTED** [review]
Only two minor code style notes, rest looks good!

File: iphone/cli/commands/_build.js (hunk 1 above)
Comment (cb1kenobi) [inline iphone/cli/commands/_build.js]: We are not introducing ES2015 syntax into the current Titanium build.

File: iphone/cli/hooks/package.js (hunk 2 above)
Comment (cb1kenobi) [inline iphone/cli/hooks/package.js]: Good catch. Fixed.

File: iphone/cli/commands/_build.js (hunk 1 above)
Comment (janvennemann) [inline iphone/cli/commands/_build.js]: Makes sense, better to avoid mix and match the different ES versions.

**Review by janvennemann — APPROVED** [review]
CR: Passed

**Pull request merged**